### ISSUE TITLE
Fresh git clone couldn't be compiled

### ISSUE THREAD

**Issue by datalot-369** [eranif/codelite#2674]
```bash
$ cmake --build . -j 2
[  1%] Built target wxsqlite3
[  2%] Built target wxcLib
[  3%] Built target PHPParser
[  3%] Built target codelite-cc
[  6%] Built target codelite_cppcheck
[  9%] Built target codelite_indexer
[  9%] Built target codelite-echo
[ 10%] Built target databaselayersqlite
[ 12%] Built target wxshapeframework
[ 23%] Built target libcodelite
[ 23%] Building CXX object Plugin/CMakeFiles/plugin.dir/EclipseSQLThemeImporter.cpp.o
[ 23%] Building CXX object Plugin/CMakeFiles/plugin.dir/EclipseRustThemeImporter.cpp.o
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp: In member function ‘virtual LexerConf::Ptr_t EclipseRustThemeImporter::Import(const wxFileName&)’:
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:16:78: error: ‘wxSTC_LEX_RUST’ was not declared in this scope
     LexerConf::Ptr_t lexer = InitializeImport(eclipseXmlFile, GetLangName(), wxSTC_LEX_RUST);
                                                                              ^~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:16:78: note: suggested alternative: ‘wxSTC_LEX_RUBY’
     LexerConf::Ptr_t lexer = InitializeImport(eclipseXmlFile, GetLangName(), wxSTC_LEX_RUST);
                                                                              ^~~~~~~~~~~~~~
                                                                              wxSTC_LEX_RUBY
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:17:24: error: ‘wxSTC_RUST_DEFAULT’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_DEFAULT, "Default", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:17:24: note: suggested alternative: ‘wxSTC_ST_DEFAULT’
     AddProperty(lexer, wxSTC_RUST_DEFAULT, "Default", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~
                        wxSTC_ST_DEFAULT
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:26:24: error: ‘wxSTC_RUST_LEXERROR’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_LEXERROR, "Lex error", errorColour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:26:24: note: suggested alternative: ‘wxSTC_RB_ERROR’
     AddProperty(lexer, wxSTC_RUST_LEXERROR, "Lex error", errorColour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
                        wxSTC_RB_ERROR
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:27:24: error: ‘wxSTC_RUST_COMMENTBLOCK’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCK, "Comment block", m_multiLineComment.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:27:24: note: suggested alternative: ‘wxSTC_AVS_COMMENTBLOCK’
     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCK, "Comment block", m_multiLineComment.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_AVS_COMMENTBLOCK
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:28:24: error: ‘wxSTC_RUST_COMMENTLINE’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_COMMENTLINE, "Comment line", m_singleLineComment.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:28:24: note: suggested alternative: ‘wxSTC_PAS_COMMENTLINE’
     AddProperty(lexer, wxSTC_RUST_COMMENTLINE, "Comment line", m_singleLineComment.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_PAS_COMMENTLINE
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:29:24: error: ‘wxSTC_RUST_COMMENTBLOCKDOC’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCKDOC, "Comment block doc", m_javadoc.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:29:24: note: suggested alternative: ‘wxSTC_AVS_COMMENTBLOCK’
     AddProperty(lexer, wxSTC_RUST_COMMENTBLOCKDOC, "Comment block doc", m_javadoc.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_AVS_COMMENTBLOCK
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:30:24: error: ‘wxSTC_RUST_COMMENTLINEDOC’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_COMMENTLINEDOC, "Comment line doc", m_javadoc.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:30:24: note: suggested alternative: ‘wxSTC_FS_COMMENTLINEDOC’
     AddProperty(lexer, wxSTC_RUST_COMMENTLINEDOC, "Comment line doc", m_javadoc.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_FS_COMMENTLINEDOC
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:31:24: error: ‘wxSTC_RUST_WORD’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD, "Word", m_keyword.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:31:24: note: suggested alternative: ‘wxSTC_BAT_WORD’
     AddProperty(lexer, wxSTC_RUST_WORD, "Word", m_keyword.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~
                        wxSTC_BAT_WORD
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:32:24: error: ‘wxSTC_RUST_WORD2’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD2, "Word2", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:32:24: note: suggested alternative: ‘wxSTC_LUA_WORD2’
     AddProperty(lexer, wxSTC_RUST_WORD2, "Word2", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD2
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:33:24: error: ‘wxSTC_RUST_WORD3’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD3, "Word3", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:33:24: note: suggested alternative: ‘wxSTC_LUA_WORD3’
     AddProperty(lexer, wxSTC_RUST_WORD3, "Word3", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD3
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:34:24: error: ‘wxSTC_RUST_WORD4’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD4, "Word4", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:34:24: note: suggested alternative: ‘wxSTC_LUA_WORD4’
     AddProperty(lexer, wxSTC_RUST_WORD4, "Word4", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD4
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:35:24: error: ‘wxSTC_RUST_WORD5’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD5, "Word5", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:35:24: note: suggested alternative: ‘wxSTC_LUA_WORD5’
     AddProperty(lexer, wxSTC_RUST_WORD5, "Word5", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD5
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:36:24: error: ‘wxSTC_RUST_WORD6’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD6, "Word6", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:36:24: note: suggested alternative: ‘wxSTC_LUA_WORD6’
     AddProperty(lexer, wxSTC_RUST_WORD6, "Word6", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD6
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:37:24: error: ‘wxSTC_RUST_WORD7’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_WORD7, "Word7", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:37:24: note: suggested alternative: ‘wxSTC_LUA_WORD7’
     AddProperty(lexer, wxSTC_RUST_WORD7, "Word7", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~
                        wxSTC_LUA_WORD7
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:38:24: error: ‘wxSTC_RUST_STRING’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_STRING, "String", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:38:24: note: suggested alternative: ‘wxSTC_ST_STRING’
     AddProperty(lexer, wxSTC_RUST_STRING, "String", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~
                        wxSTC_ST_STRING
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:39:24: error: ‘wxSTC_RUST_STRINGR’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_STRINGR, "String raw", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:39:24: note: suggested alternative: ‘wxSTC_ST_STRING’
     AddProperty(lexer, wxSTC_RUST_STRINGR, "String raw", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~
                        wxSTC_ST_STRING
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:40:24: error: ‘wxSTC_RUST_CHARACTER’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_CHARACTER, "Character", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:40:24: note: suggested alternative: ‘wxSTC_ST_CHARACTER’
     AddProperty(lexer, wxSTC_RUST_CHARACTER, "Character", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~
                        wxSTC_ST_CHARACTER
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:41:24: error: ‘wxSTC_RUST_OPERATOR’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_OPERATOR, "Operator", m_oper.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:41:24: note: suggested alternative: ‘wxSTC_SN_OPERATOR’
     AddProperty(lexer, wxSTC_RUST_OPERATOR, "Operator", m_oper.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
                        wxSTC_SN_OPERATOR
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:42:24: error: ‘wxSTC_RUST_IDENTIFIER’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_IDENTIFIER, "Identifier", m_variable.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:42:24: note: suggested alternative: ‘wxSTC_PAS_IDENTIFIER’
     AddProperty(lexer, wxSTC_RUST_IDENTIFIER, "Identifier", m_variable.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~
                        wxSTC_PAS_IDENTIFIER
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:43:24: error: ‘wxSTC_RUST_LIFETIME’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_LIFETIME, "Lifetime", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:43:24: note: suggested alternative: ‘wxSTC_ASM_DIRECTIVE’
     AddProperty(lexer, wxSTC_RUST_LIFETIME, "Lifetime", m_foreground.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~
                        wxSTC_ASM_DIRECTIVE
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:44:24: error: ‘wxSTC_RUST_MACRO’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_MACRO, "Macro", wxColour(m_klass.colour).GetAsString(wxC2S_HTML_SYNTAX),
                        ^~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:44:24: note: suggested alternative: ‘wxSTC_AU3_MACRO’
     AddProperty(lexer, wxSTC_RUST_MACRO, "Macro", wxColour(m_klass.colour).GetAsString(wxC2S_HTML_SYNTAX),
                        ^~~~~~~~~~~~~~~~
                        wxSTC_AU3_MACRO
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:46:24: error: ‘wxSTC_RUST_BYTESTRING’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_BYTESTRING, "Byte string", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:46:24: note: suggested alternative: ‘wxSTC_ST_STRING’
     AddProperty(lexer, wxSTC_RUST_BYTESTRING, "Byte string", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~
                        wxSTC_ST_STRING
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:47:24: error: ‘wxSTC_RUST_BYTESTRINGR’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_BYTESTRINGR, "Byte string raw", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:47:24: note: suggested alternative: ‘wxSTC_ST_STRING’
     AddProperty(lexer, wxSTC_RUST_BYTESTRINGR, "Byte string raw", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_ST_STRING
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:48:24: error: ‘wxSTC_RUST_BYTECHARACTER’ was not declared in this scope
     AddProperty(lexer, wxSTC_RUST_BYTECHARACTER, "Byte character", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~
/home/datalot/git-clones/codelite/Plugin/EclipseRustThemeImporter.cpp:48:24: note: suggested alternative: ‘wxSTC_ST_CHARACTER’
     AddProperty(lexer, wxSTC_RUST_BYTECHARACTER, "Byte character", m_string.colour, m_background.colour);
                        ^~~~~~~~~~~~~~~~~~~~~~~~
                        wxSTC_ST_CHARACTER
make[2]: *** [Plugin/CMakeFiles/plugin.dir/build.make:635: Plugin/CMakeFiles/plugin.dir/EclipseRustThemeImporter.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:314: Plugin/CMakeFiles/plugin.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

**Comment by dghart** [comment]
Works-for-me. Was yours a clean build? If not, that's probably the problem; rust is a new addition.

BTW, an easy way to build, and clean, is to use a subdirectory. For example (for a debug build against a gtk+3 wxWidgets install):
(mkdir -p build-dbg3 && cd build-dbg3/ && make clean && rm -f CMakeCache.txt &&  cmake -G "Unix Makefiles" VERBOSE=1 -DCMAKE_BUILD_TYPE=DebugFull -DCL_PREFIX=$(pwd)/../CLgtk3-install .. && make -j`nproc` && make install)

**Comment by datalot-369** [comment]
I just have followed the Build Step for the first time. Let me try yours.

**Comment by datalot-369** [comment]
I just have followed the Build Step for the first time. Let me try yours.

**Comment by datalot-369** [comment]
I stuck with the same issue at the `EclipseRustThemeImporter.cpp`, here is the pipeline in a whole fresh git clone:

[procedure.txt](https://github.com/eranif/codelite/files/5991263/procedure.txt)

**Comment by dghart** [comment]
Ah, I see: you're building against wx3.0.4 and that hasn't heard of wxSTC_LEX_RUST etc; it's new in wx3.1.

That needs fixing in CodeLite, of course. Thank you for reporting it. Meanwhile, if you wish to build from git HEAD  you could install wx3.1.4 (perhaps from https://wiki.codelite.org/pmwiki.php/Main/WxWidgets31Binaries, depending on your distro) or build it yourself. Otherwise you could revert to CodeLite 15.0.

**Comment by datalot-369** [comment]
@dghart Thank you for your help!

I just forgot to include my OS decription:
```
- MX Linux 19.3 amd64 (Basically Debian Buster)
- XFCE 4.14
- GTK 3.24
- And now, libwxgtk3.1-dev 3.1.4-1.buster
```

(You can ignore this) The program has successfully compiled and started showing me a First-Install window for appearance configurations (I can report some things about it), while after its automatic restart it has displayed a pretty crash (?) window. Below is the backtrace, but I will triage it anyways. **Thank you for helping me compiling Codelite!**

```
ASSERT INFO:
../src/common/event.cpp(1926): assert "nNew != dynamicEvents.size()" failed in SearchDynamicEventTable().

BACKTRACE:
[1] wxEvtHandler::SearchDynamicEventTable(wxEvent&)
[2] wxEvtHandler::TryHereOnly(wxEvent&)
[3] wxEvtHandler::ProcessEventLocally(wxEvent&)
[4] wxEvtHandler::ProcessEvent(wxEvent&)
[5] EventNotifier::ProcessEvent(wxEvent&)
[6] wxEvtHandler::ProcessPendingEvents()
[7] wxAppConsoleBase::ProcessPendingEvents()
[8] wxApp::DoIdle()
[9] g_main_context_dispatch
[10] g_main_loop_run
[11] gtk_main
[12] wxGUIEventLoop::DoRun()
[13] wxEventLoopBase::Run()
[14] wxAppConsoleBase::MainLoop()
[15] wxEntry(int&, wchar_t**)
[16] main
[17] __libc_start_main
[18] _start
```

**Comment by dghart** [comment]
This should now be fixed by #a10276df1
